Convert timemodel to time class

kiwibang opened the pull request:
- This PR changes the name of TimeModel into Time

- Renamed so that it is consistent with the other objects 
in the application (e.g. Interview, Rating, Applicant, etc)

- Refactored time-related methods for better separation of concerns

- closes #113

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T11-2/tp

Commit:
convert TimeModel to Time class

- this commit changes the name of "TimeModel" class
to "Time"

- Besides syntactic changes, the Interview class now
stores the start and end time as Time objects

- Refactored the Time-related methods into the Time class,
for better Separation of Concerns

jingjie88 commented on src/main/java/seedu/address/model/interview/Interview.java:
```diff
@@ -149,19 +147,9 @@ public boolean isSameInterview(Interview otherInterview) {
-                && (time.isBefore(WORK_END) || time.equals(WORK_END));
+                && startTime.isWithinWorkingHours()
+                && endTime.isWithinWorkingHours()
+                && startTime.getTime().equals(endTime.getTime());
```
I think this line should check if the dates are the same and not the time. The start time and end time are valid only if the date for both are the same.

kiwibang replied:
Good catch! My apologies for the carelessness! 👍 
Adding commit to this PR branch to fix it right now.

jingjie88 reviewed (approved):
LGTM! Just one minor thing to fix.

Commit:
Fix small bug in Interview::isValid

- Fix isValid checking time to isValid checking date, of the
start time and end time of interview

jingjie88 reviewed (approved):
LGTM!
